core-android/src/main/java/com/uber/sdk/android/core/auth/LoginActivity.java

```diff
@@ -152,28 +190,44 @@ protected void init() {
     }
 
     protected void loadUrl() {
-        sessionConfiguration = (SessionConfiguration) getIntent().getSerializableExtra(EXTRA_SESSION_CONFIGURATION);
-        if (sessionConfiguration == null) {
-            onError(AuthenticationError.UNAVAILABLE);
+        Intent intent = getIntent();
+
+        sessionConfiguration = (SessionConfiguration) intent.getSerializableExtra(EXTRA_SESSION_CONFIGURATION);
+        responseType = (ResponseType) intent.getSerializableExtra(EXTRA_RESPONSE_TYPE);
+
+        if (!validateRequestParams()) {
             return;
         }
 
-        if ((sessionConfiguration.getScopes() == null || sessionConfiguration.getScopes().isEmpty())
-                && (sessionConfiguration.getCustomScopes() == null  || sessionConfiguration.getCustomScopes().isEmpty())) {
-            onError(AuthenticationError.INVALID_SCOPE);
+        String redirectUri = sessionConfiguration.getRedirectUri() != null ? sessionConfiguration
+                .getRedirectUri() : getApplicationContext().getPackageName() + "uberauth";
```

Comment:
Is this the correct fallback behavior if the redirect uri is null? It seems like from other places in the code and the readme that we would want `getApplicationContext().getPackageName().concat(".uberauth://redirect")` here instead, but I wasn't sure.

Reply:
getApplicationContext().getPackageName().concat(".uberauth://redirect") would be correct in this case, however redirectUri must be set before calling LoginManager.login(), as it does the validation. 

While we have the autogeneration for the handler activity, we don't yet automatically try to assemble the redirectUri when loading the URI and that should probably be a separate conversation.

Reply:
Can you clarify what you mean by the 2nd paragraph here? Do you mean that we don't assemble the redirectUri when we load a url in the chrometab?

Reply:
Disregard the second paragraph, I missed the refactor to move this up. My original thought was about handling the error condition of fallbacks when the user doesn't set the URI (Which was gated by the legacy URI handler earlier I think).

However, it does seem to have a type on the fallback clause and should switch to my first recommendation.